pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -112,14 +113,29 @@ func (s *Server) initCertController(args *PilotArgs) error {
 //
 // TODO: If the discovery address in mesh.yaml is set to port 15012 (XDS-with-DNS-certs) and the name
 // matches the k8s namespace, failure to start DNS server is a fatal error.
-func (s *Server) initDNSCerts(hostname, namespace string) error {
+func (s *Server) initDNSCerts(hostname, customHostname, namespace string) error {
```

Comment:
Since both parameters hostname and customHostname will be placed in the names array (e.g., names := []string{hostname}), to simplify the parameters of the function initDNSCerts(hostname, customHostname, namespace string), the function can be declared as initDNSCerts(hostname []string, namespace string).

Reply:
Actually it is not treated the same, the hostname part 1 are used later when creating CSR thus it is simpler when not merging them into an array.